### PR TITLE
Feature/79 add GitHub actions to build and test project

### LINKED ISSUE #79: Add GitHub actions to build and test project

### PR DESCRIPTION
**Pull request by NoB0**
Workflow is inspired by DialogueKit
Additions:
- CI
- Merge action
- Update documentation action to build versioned documentation
- Add `requirements.txt`

Depends on #80 
Fixes #79

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+pre-commit
+flake8
+black
+pytest
+pytest-cov
+mypy
+docformatter
+types-PyYAML
+types-requests
+dialoguekit
```
Comment (kbalog) [inline requirements/requirements.txt:10]: Should we require a specific minimum version here?

**Review by kbalog — APPROVED** [review]
LGTM, see question

File: requirements/requirements.txt (hunk 1 above)
Comment (NoB0) [reply requirements/requirements.txt:10]: I was thinking about it as well. We could just to be safe and update the version if necessary. As long as the structure of DialogueKit stays the same it should work fine by just downloading the last version on pypi.

Comment (kbalog) [reply requirements/requirements.txt:10]: OK, let's add a minumum version then and we update as necessary.

Comment (NoB0) [reply requirements/requirements.txt]: Done.

**Pull request merged**